### PR TITLE
Fix trash (low-quality code) in AWS S3

### LINKED ISSUE #52771: Creating view on S3 glob pattern takes unexpectedly long time
**Describe the unexpected behaviour**

When creating a view, I am observing 4 seconds from my laptop in us east to data in us west, when I'd expect it to be instant.

**How to reproduce**
* Which ClickHouse server version to use

`ClickHouse server version 23.7.1 revision 54464`

* Queries to run that lead to unexpected result

```
CREATE VIEW s3_parquet AS
SELECT *
FROM s3('https://s3.us-west-2.amazonaws.com/ookla-open-data/parquet/performance/type=*/year=*/quarter=*/*.parquet', 'Parquet', 'quadkey Nullable(String), tile Nullable(String), avg_d_kbps Nullable(Int64), avg_u_kbps Nullable(Int64), avg_lat_ms Nullable(Int64), tests Nullable(Int64), devices Nullable(Int64)')

Query id: 7eaf74c7-25ba-4748-8dff-6f4accdcdabb

Ok.

0 rows in set. Elapsed: 4.021 sec.

d2e52eb1a093 :) select count() from s3_parquet

SELECT count()
FROM s3_parquet

Query id: a703f818-ac38-40ea-a1fa-8736c28f5bd3

┌───count()─┐
│ 185832935 │
└───────────┘

1 row in set. Elapsed: 40.022 sec. Processed 185.83 million rows, 7.17 GB (4.64 million rows/s., 179.10 MB/s.)
```

Without View (killed to reset cache):
```
SELECT count()
FROM s3('https://s3.us-west-2.amazonaws.com/ookla-open-data/parquet/performance/type=*/year=*/quarter=*/*.parquet', 'Parquet', 'quadkey Nullable(String), tile Nullable(String), avg_d_kbps Nullable(Int64), avg_u_kbps Nullable(Int64), avg_lat_ms Nullable(Int64), tests Nullable(Int64), devices Nullable(Int64)')

Query id: 2f8b2625-9e89-418e-8d58-e5cff0d2e7e2

┌───count()─┐
│ 185832935 │
└───────────┘

1 row in set. Elapsed: 40.336 sec. Processed 185.83 million rows, 7.17 GB (4.61 million rows/s., 177.71 MB/s.)
```

I've specified the schema to remove the possibility that it needs to scan files to determine schema. As you can see creating the view takes 4 seconds, but the count of both queries takes nearly identical time. I am not sure what it is doing when creating the view but from at least this query it does not seem to be helpful.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Speed up requests to S3 when a profile is not set, credentials are not set, and IMDS is not available (for example, when you are querying a public bucket on a machine outside of a cloud). This closes #52771.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 6f0f27838745d427966a3c949316a541d9fda7ac with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68082/6f0f27838745d427966a3c949316a541d9fda7ac/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67940/eb4ea0757730f8b8e59b9230d72aea3ca4bb1ff3/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/IO/S3/Credentials.cpp [hunk 1]
```diff
@@ -785,11 +785,11 @@ S3CredentialsProviderChain::S3CredentialsProviderChain(
 
             /// EC2MetadataService throttles by delaying the response so the service client should set a large read timeout.
```
Comment (jkartseva) [inline src/IO/S3/Credentials.cpp:786]: The comment should elaborate on why the connection timeout is small.

**Review by jkartseva — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Test draft:

```
AWS_EC2_METADATA_SERVICE_ENDPOINT='https://10.255.255.255/' ch --time -q "SELECT * FROM s3('https://google.com/test')"
```

(we will have to check that at least once it responds in less than a second)

**Comment by alexey-milovidov** [comment]
Needs https://github.com/ClickHouse/ClickHouse/issues/67184

**Pull request merged**